File: camera.py
import cv2

#get the camera
class CameraRead:
    def __init__(self, camera_index = 2):
        self.cap = cv2.VideoCapture(2)
        #set desired camera params
        self.cap.set(cv2.CAP_PROP_FRAME_WIDTH, 960)
        self.cap.set(cv2.CAP_PROP_FRAME_HEIGHT, 450)
        self.cap.set(cv2.CAP_PROP_FPS, 60)

    #method to retrieve images in integrate
    def get_img(self):
        ret, frame = self.cap.read()

        if ret is None:
            print("CAMERA ERROR: no image retrieved")
            return False, None
        
        print("IMG FOUND")
        return ret, frame
    
    #method to release webcam at runtime close
    def release(self):
        self.cap.release()

#single testing
#comment out once going to int
if __name__ == "__main__":
    cam = CameraRead()
    ret, frame = cam.get_img()

File: src/camera.cpp
#include "lab_08.h"

class CameraRead{
    private:
        cv::VideoCapture cap{};

    public: 
        CameraRead(int camera_index) : cap(camera_index){
            //set desired camera params
            cap.set(cv::CAP_PROP_FRAME_WIDTH, 960);
            cap.set(cv::CAP_PROP_FRAME_HEIGHT, 450);
            cap.set(cv::CAP_PROP_FPS, 60);
        
            // Check if the webcam was opened successfully
            if (!cap.isOpened()) {
                std::cout << "Error: Could not access the webcam." << std::endl;
            } else {
                std::cout << "Webcam accessed successfully!" << std::endl;
            }
        }

        ~CameraRead{
            if (cap.isOpened()) {
            cap.release();
            std::cout << "Camera released safely." << std::endl;
            }
        }

        bool GetFrame(cv::Mat &frame) {

            //retrieve new frame when requested
            bool ret = cap.read(frame);
            
            //if no frame is captured return with error
            if (!ret) {
                std::cout << "Error retrieving frame" <<std::endl;
                return ret;
            }

            return ret;
        }
};
